samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/authz/integrated/AuthzQueriesFacade.java

```diff
@@ -299,4 +301,16 @@ public boolean checkMembership(String siteId) {
     return isMember;
   }
 
+    public void hardDeleteAuthzData(String agentId) {
+      Session session = getHibernateTemplate().getSessionFactory().getCurrentSession();
+        try {
+          session.createQuery("delete from AuthorizationData where agentId = :id")
+              .setString("id", agentId)
+              .executeUpdate();
+          session.flush();
```

Comment:
are you sure the flush is needed?

Reply:
I did made an update, which uses criteria builder (:
Hope I did not miss something